libyul/Object.cpp

```diff
@@ -144,33 +145,62 @@ set<YulString> Object::qualifiedDataNames() const
 	return qualifiedNames;
 }
 
-vector<size_t> Object::pathToSubObject(YulString _qualifiedName) const
+void Object::traverseObjectTree(Object const* _object, YulString _qualifiedName, std::function<bool(Object const*)> _visitor)
 {
-	yulAssert(_qualifiedName != name, "");
-	yulAssert(subIndexByName.count(name) == 0, "");
-
-	if (boost::algorithm::starts_with(_qualifiedName.str(), name.str() + "."))
-		_qualifiedName = YulString{_qualifiedName.str().substr(name.str().length() + 1)};
 	yulAssert(!_qualifiedName.empty(), "");
 
 	vector<string> subObjectPathComponents;
 	boost::algorithm::split(subObjectPathComponents, _qualifiedName.str(), boost::is_any_of("."));
 
-	vector<size_t> path;
-	Object const* object = this;
-	for (string const& currentSubObjectName: subObjectPathComponents)
+	for (size_t i = 0; i < subObjectPathComponents.size(); i++)
```

Comment:
Why not range-based for?

```suggestion
	for (auto const& currentSubObjectName: subObjectPathComponents)
```

Reply:
No specific reason. Fixed :)